src/main.cpp

```diff
@@ -154,15 +155,15 @@ vector<TT_Entry> transposition_table;
     str += '1' + (flip ? (7 - move.from / 8) : (move.from / 8));
     str += 'a' + (move.to % 8);
     str += '1' + (flip ? (7 - move.to / 8) : (move.to / 8));
-    if (move.promo != None) {
-        str += "\0nbrq\0\0"[move.promo];
+    if (move.promo) {
+        str += "\0\0nbrq\0"[move.promo];
     }
     return str;
 }
 
 [[nodiscard]] int piece_on(const Position &pos, const int sq) {
     const u64 bb = 1ULL << sq;
-    for (int i = 0; i < 6; ++i) {
+    for (int i = 1; i < 7; ++i) {
```

Comment:
Can't this run from 0 to 7? (maybe with a compiled out assert for the fallthrough)